src/Microsoft.OData.Core/UriParser/ODataQueryOptionParser.cs

```diff
@@ -533,8 +553,9 @@ private bool TryGetQueryOption(string queryOptionName, out string value)
                 return true;
             }
 
-            throw new ODataException(Strings.QueryOptionUtils_QueryParameterMustBeSpecifiedOnce(queryOptionName));
+            throw new ODataException(Strings.QueryOptionUtils_QueryParameterMustBeSpecifiedOnce(
+                isNoDollarQueryOptionsEnabled ? string.Format(CultureInfo.InvariantCulture, "${0}/{0}", noDollarQueryOptionName ?? string.Empty) : trimmedQueryOptionName));

```

Comment:
if isNoDollarQueryOptionsEnabled is true and noDollarQueryOptionName is null, we should still use trimmedQueryOptionName, right?